unit_tests/test_events.py

```diff
@@ -338,3 +342,28 @@ def test_failed_stall_during_filter(self):
                                line="[99.80.124.204] [stdout] Mar 31 09:08:10 warning|  reactor stall 20").publish()
 
         self.assertEqual(event.severity, Severity.DEBUG)
+
+    @pytest.mark.parametrize("test_input,expected", [(None, ''), (2, '2s'), (2.52, '2.52s'), (326, 326),
+                                                     (4598, '1h16m38s'), (87400, '1h16m38s')])
+    @staticmethod
+    def test_duration_format(test_input, expected):
+        event = NodetoolEvent(nodetool_command="scrub --skip-corrupted drop_table_during_repair_ks_0", node='1.0.0.121',
+                              options="more options", publish_event=False)
+
+        event.duration = test_input
+        assert event.duration_formatted, expected
+
+    @mock.patch('sdcm.sct_events.base.SctEvent.publish')
+    def test_publish_called(self, publish):
+        event = NodetoolEvent(nodetool_command="scrub", node='1.0.0.121',
+                              options="", publish_event=True)
+        event.begin_event()
+        self.assertTrue(publish.called, "Publish function was not called unexpectedly")
+
+    @mock.patch('sdcm.sct_events.base.SctEvent.publish')
+    def test_publish_not_called(self, publish):
+        event = NodetoolEvent(nodetool_command="scrub", node='1.0.0.121',
+                              options="", publish_event=True)
+        event.publish_event = False
```

Comment:
I expected following to be here:

```
        event = NodetoolEvent(nodetool_command="scrub", node='1.0.0.121', options="", publish_event=False)
```
"False" be provided directly as "publish_event" value, not being updated later.

Reply:
fixed